src/averageAges.js

```diff
@@ -37,7 +44,25 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const arrayWithMothers = people
+    .filter((person) => {
+      return person.mother;
+    })
+    .map((person) => {
+      return person.mother;
+    });
+  const arrayWithYears = people
+    .filter((person) => {
+      return withChildren ? person.sex === 'f'
+      && arrayWithMothers.includes(person.name) : person.sex === 'f';
+    })
```

Comment:
we can simplify this by checking whether in the array of people there are those who have `.mother === .name`
it should look like calculateMenAverageAge by lines of code.

PS. Try to use `.some`